.github/workflows/composites/upload-docker-images/action.yaml

```diff
@@ -0,0 +1,15 @@
+# we are not going to use upload/download actions, since it is slower than cache usage
+# as key to the cache we are going to use DOCKER_IMAGES_KEY, which is the same as GITHUB_RUN_ID
+
+name: upload docker images
+description: upload docker images
+runs:
+  using: "composite"
+  steps:
```

Comment:
circleci, unlike github actions, has a common path for the entire workflow, so for example: 

- I can put images into `/tmp/docker/images` in one step
- reference these images via `docker load -i` in any step I would like

github actions does not have such support, they recommend to use upload/download actions. It turns out that uploading our images, currently takes 15 minutes, far too much.

So instead, I am using cache with a slight hack - using `DOCKER_IMAGES_KEY` (`echo "DOCKER_IMAGES_KEY=$(echo $GITHUB_RUN_ID)" >> $GITHUB_ENV`) that I set in `env-variables` composite action and use that. Since  `GITHUB_RUN_ID` is unique per workflow, this achieves the "upload/download" without respective actions, but using `cache` action instead. It shrinks this step to 1 minute.